modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -254,6 +259,18 @@ class Session implements ISession {
 
     private boolean statsEnabled
 
+    private volatile boolean cidEnabled
+
+    boolean getCidEnabled() { cidEnabled }
+
+    private HashCode executionHash
+
+    private CidStore cidStore
+
+    CidStore getCidStore() { cidStore }
+
+    String getExecutionHash() { executionHash }
+
```

Comment:
I wonder if most of those could not be isolated in the CidObserver or a dedicated class, ideally here there should be only a reference to the observer

Reply:
I have moved the execution hash to the observer, however I have kept the cidStore because I get the store from the session in the CmdCid class.